curl_cffi/requests/session.py

```diff
@@ -279,29 +269,31 @@ def _set_curl_options(
         c = curl
 
         # method
-        if method == "POST":
-            c.setopt(CurlOpt.POST, 1)
-        elif method != "GET":
+        if method != "GET":
             c.setopt(CurlOpt.CUSTOMREQUEST, method.encode())
+            if method == "POST":
```

Comment:
For POST, setting `CUSTOMREQUEST` can be problematic. A common scenario is that user submits a form via POST, and the server redirects to a page that you need to GET. `CurlOpt.POST` is actually made for that, while `CUSTOMREQUEST` will send **two** POST requests.